ipaserver/install/custodiainstance.py

```diff
@@ -40,16 +39,18 @@ def __config_file(self):
         template_file = os.path.basename(self.config_file) + '.template'
         template = os.path.join(paths.USR_SHARE_IPA_DIR, template_file)
         httpd_info = pwd.getpwnam(constants.HTTPD_USER)
-        sub_dict = dict(IPA_CUSTODIA_CONF_DIR=paths.IPA_CUSTODIA_CONF_DIR,
-                        IPA_CUSTODIA_SOCKET=paths.IPA_CUSTODIA_SOCKET,
-                        IPA_CUSTODIA_AUDIT_LOG=paths.IPA_CUSTODIA_AUDIT_LOG,
-                        LDAP_URI=installutils.realm_to_ldapi_uri(self.realm),
-                        UID=httpd_info.pw_uid, GID=httpd_info.pw_gid)
+        sub_dict = dict(
+            IPA_CUSTODIA_CONF_DIR=paths.IPA_CUSTODIA_CONF_DIR,
+            IPA_CUSTODIA_KEYS=paths.IPA_CUSTODIA_KEYS,
+            IPA_CUSTODIA_SOCKET=paths.IPA_CUSTODIA_SOCKET,
+            IPA_CUSTODIA_AUDIT_LOG=paths.IPA_CUSTODIA_AUDIT_LOG,
+            LDAP_URI=installutils.realm_to_ldapi_uri(self.realm),
+            UID=httpd_info.pw_uid,
+            GID=httpd_info.pw_gid
+        )
         conf = ipautil.template_file(template, sub_dict)
-        fd = open(self.config_file, "w+")
-        fd.write(conf)
-        fd.flush()
-        fd.close()
+        with open(self.config_file, "w") as f:
+            f.write(conf)
```

Comment:
unless "with open" guarantees there is a flush you need to retain f.flush()

Reply:
Yes, it does. close also flushes all buffers. Or did you mean to fsync / fdatasync the files from Kernel cache to disk?

Reply:
I thought flush() causes a fsync/fdatasync, if it doesn't that's a mistake we need to fix here and elsewhere and add that.
We want to be sure keys are stored on disk before we continue here.

Reply:
No, ``file.flush()`` is equivalent to ``fflush(3)``. It only flushes the internal buffer to the file descriptor. I can add fsync().

Reply:
Yeah, I really think the flush() here has been used in error, and fsync() was meant to be used instead. That flush_fsync is super neat. Thanks.